### PR TITLE
xhr: Use .{any,window}.js syntax for abort-* tests

### PR DESCRIPTION
**Pull request by CYBAI**
Part of #11588.

Btw, after reading [documentation of `.any.js`](https://web-platform-tests.org/writing-tests/testharness.html#multi-global-tests), it seems there's no syntax like `// META: help=https://xxx.spec.whatwg.org` 👀? Somehow, I wonders the help spec links in the `htm` tests might be helpful 🤔

cc @zcorpan

### CODE REVIEW

**Comment by CYBAI** [comment]
Btw, for [the `abort-event-order.htm` test](https://github.com/web-platform-tests/wpt/blob/master/xhr/abort-event-order.htm), I tried to make it into `.any.js` with following `META`.

```js
// META: title=XMLHttpRequest: The abort() method: abort and loadend events
// META: script=resources/xmlhttprequest-event-order.js
```

however, I will get `prepare_xhr_for_event_order_test is not defined` 🤔 so I didn't update it to `.any.js` syntax yet.

**Comment by annevk** [comment]
The reindenting makes this really hard to follow. Without that GitHub would see it as a move+change. Whereas now it's removal+addition, which basically requires going line-by-line. Arguably a bug in tooling, but would it be possible to make these changes without changing indentation or at least doing that in a separate commit?

You'll also want to add a single newline at the end of resources. That keeps diffs more readable going forward.

**Comment by CYBAI** [comment]
> The reindenting makes this really hard to follow. Without that GitHub would see it as a move+change. Whereas now it's removal+addition, which basically requires going line-by-line. Arguably a bug in tooling, but would it be possible to make these changes without changing indentation or at least doing that in a separate commit?
> 
> You'll also want to add a single newline at the end of resources. That keeps diffs more readable going forward.

Yeah, agree with you. Due to installing a VSCode extension, it causes the re-indentation; I just found which the extension is and remove it. I think I can update the file without re-indentation now! Let me update them! Thanks!

**Comment by CYBAI** [comment]
@annevk I only re-indent the tests without leading spaces but I found `git` still see them as `removal+addition` so I separated each file as a commit. 👀

**Comment by annevk** [comment]
Did you use `git mv` followed by only removing the HTML and not changing the indentation of the script? This seems rather strange.

**Comment by jgraham** [comment]
Note that git doesn't record moves explictly, so unlike `hg` whether or not you use `git mv` doesn't matter; if the files are similar enough it will regard them as moves, if not it won't. I don't recall the exact details of the lagorithm it uses, but possibly changing whitespace on every line would be enough to make it look like a non-move.

**Comment by CYBAI** [comment]
> Did you use `git mv` followed by only removing the HTML and not changing the indentation of the script? This seems rather strange.

No, I didn't use `git mv` and I just tried to use it and then remove the leading spaces, it will still be `removal+addition` so I think it works like @jgraham said.

For `remove the leading spaces`, I mean, for example,

from

```js
    var test = async_test();

    test.step(function()
    {
```

to

```js
var test = async_test();

test.step(function()
{
```

**Comment by zcorpan** [comment]
A workaround would be to have the move without any content changes in one commit, and the content changes in a separate commit.

> btw, in this case, would you think it's fine to re-indent them and move some unnecessary test variable to just async_test ?

I think that would be good.

**Comment by annevk** [comment]
Okay, so the main problem here is the removal of leading spaces, let's not do that. Removing the HTML should be fine. At least that approach has worked before.

**Comment by CYBAI** [comment]
@annevk @zcorpan Understood. Would you think it's fine to only remove the HTML in this PR? Or, it's okay to remove the HTML in a commit and then re-indent them in separate commits? Thanks! 🙇

**Comment by annevk** [comment]
Yeah, I'd couple moving the file and removing the HTML in a single commit. That I've seen work before and should continue to work.

Then I'd remove leading spaces in a separate commit if we wanted to do that. (We could also leave it and only do it if those resources need touching for some other reason.)

**Comment by CYBAI** [comment]
@annevk @zcorpan I updated most of the `abort-*` tests from `.htm` to `.{any,window}.js` syntax without removing leading spaces and `git` can recognize them as `moved` files. However, there're 3 tests recognized `addition+removal` even I don't remove leading spaces. So, I make 3 of them as separate commits and remove the leading spaces. Please let me know if there's better solution for it! Thanks! 🙇

**Comment by Ms2ger** [comment]
> However, there're 3 tests recognized addition+removal even I don't remove leading spaces. So, I make 3 of them as separate commits and remove the leading spaces.

I would suggest not doing this. For `abort-event-listeners`, the similarity index is 43%, while git's default cutoff for detecting a rename is 50%, so this is indeed what you'll see. However, on the command line, you could pass `--find-renames=43`, which will find the connection—but not if you also reindent.

**Comment by CYBAI** [comment]
> However, on the command line, you could pass --find-renames=43, which will find the connection—but not if you also reindent.

Thank you! I didn't know that 🙇  Will update them!

**Comment by CYBAI** [comment]
@Ms2ger hmm, I tried to use `--find-renames` but I can only see the diff shown as `renamed` when I use `git diff --find-renames=43`. After I add them and commit them, they're still shown as `additional+removal`. Did I miss something to let git know they're renames? 🤔

Btw, I just update the original last 3 commits into 4beed8a which only updates these 3 tests from `.htm` to `.any.js`. Thanks 🙇

**Review by Ms2ger — APPROVED** [review]
Thanks!
